Sources/PackageModel/SwiftSDKBundle.swift

```diff
@@ -136,26 +136,35 @@ public struct SwiftSDKBundle {
         _ archiver: some Archiver,
         _ observabilityScope: ObservabilityScope
     ) async throws {
-        _ = try await withTemporaryDirectory(
-            removeTreeOnDeinit: true
-        ) { temporaryDirectory in
+        let bundleName = try await withTemporaryDirectory(removeTreeOnDeinit: true) { temporaryDirectory in
             let bundlePath: AbsolutePath
 
             if
                 let bundleURL = URL(string: bundlePathOrURL),
                 let scheme = bundleURL.scheme,
                 scheme == "http" || scheme == "https"
             {
-                let bundleName = bundleURL.lastPathComponent
+                let bundleName: String
+                let fileNameComponent = bundleURL.lastPathComponent
+                if fileNameComponent.hasSuffix(".tar.gz") {
+                    bundleName = fileNameComponent
+                } else {
+                    bundleName = "bundle.tar.gz"
+                }
                 let downloadedBundlePath = temporaryDirectory.appending(component: bundleName)
 
                 let client = HTTPClient()
-                var request = HTTPClientRequest.download(
+
+                var request = HTTPClientRequest.init(method: .head, url: bundleURL)
+                request = HTTPClientRequest.download(
                     url: bundleURL,
                     fileSystem: fileSystem,
                     destination: downloadedBundlePath
                 )
                 request.options.validResponseCodes = [200]
+
+                print("Downloading a Swift SDK bundle archive from `\(bundleURL)`...")
```

Comment:
left over?

Reply:
I've kept this intentionally. As SDKs are quite large it may take a minute or more to download. This may appear as if the installation command is hanging as we don't display any download progress and there's no other output preceding this `print` statement. With it present now at least we let the user now that the download is in progress.

Reply:
use observably instead. we do not use print outside the CLI tool handlers

Reply:
This is addressed now, I've added a new `enum Output` in `SwiftSDKBundleStore`, and values of that new type are passed to the `outputHandler` closure. `print` is now only used at the top level.